### PR TITLE
Interview waitingroom dialogs with socket

### PR DESCRIPTION
**Pull request by hassanfarahani**
In this PR, I integrated the socket with authentication. In other words, if the user successfully logged in, the socket will be initialized automatically (I also defined a middleware for socket to check if the user has the valid token in the req header). After that, user redirects to the Dashboard page, over there, user can create an interview and the interview will add to the upcoming interviews in the dashboard page and then the waiting room dialog pops up. In the waiting room dialog, there is a link, which can be shared with other participants (interviewee or interviewer). As soon as the user click on the start button in the waiting room, an interview room will be created using socket. other participants can use the shared link to enter the created interview room.

### CODE REVIEW

File: client/src/pages/Dashboard.js [hunk 1]
```diff
@@ -47,29 +51,43 @@ const pastInterviews = [
...
-					align="center"
-				>
-					Past practice interviews
-				</Typography>
-				<PastInterviews rows={pastInterviews} />
-			</Box>
-		</Container>
+		<>
+			<Container maxWidth="lg">
+				<button onClick={() => setOpen(true)}>Start</button>
```
Comment (dproc96) [inline client/src/pages/Dashboard.js]: Can you update this to more closely match the styling for the rest of the app (using Material UI button)? Maybe changing the text to something like Create Interview as well

**Review by dproc96 — COMMENTED** [review]
UI looks pretty good, had one comment about that. I did noticed quite a bit of commented out code and several warnings in the console, can you address those? Also it looks like the indent  and semicolon use is not aligned between some of your files, can you run prettier to clean those up?

**Comment by bonnieli** [comment]
@hassanfarahani please update this PR now that there is new changes in `dev` :)

File: client/src/pages/Dashboard.js [hunk 2]
```diff
@@ -1,75 +1,88 @@
...
-					align="center"
-				>
-					Past practice interviews
-				</Typography>
-				<PastInterviews rows={pastInterviews} />
-			</Box>
-		</Container>
+		<>
+			<Container maxWidth="lg">
+				<Box mt={4} align='center'>
```
Comment (bonnieli) [inline client/src/pages/Dashboard.js]: notice on line 44 it's double quotes and this is single quote

File: client/src/App.js [hunk 3]
```diff
@@ -1,24 +1,28 @@
...
-import FAQ from './pages/FAQ';
-import Blog from './pages/Blog';
 import Profile from './pages/Profile';
-import Home from "./pages/HomePage";
-import Interview from './pages/Interview'
-import axios from 'axios';
-import { UserProvider } from './context/UserContext';
+import Interview from './pages/Interview';
+import Navbar from './components/layout/Navbar';
+import FAQ from './pages/FAQ'
```
Comment (bonnieli) [inline client/src/App.js]: missing semi colon

**Review by bonnieli — APPROVED** [review]

**Pull request merged**